.evergreen/run-oidc-tests.sh

```diff
@@ -2,14 +2,34 @@
 set -o errexit  # Exit the script with error if any of the commands fail
 set -o xtrace   # Write all commands first to stderr
 
+PROVIDER_NAME=${PROVIDER_NAME:-"aws"}
+PROJECT_DIRECTORY=${PROJECT_DIRECTORY:-"."}
 source "${PROJECT_DIRECTORY}/.evergreen/init-nvm.sh"
 
 MONGODB_URI=${MONGODB_URI:-"mongodb://127.0.0.1:27017"}
-MONGODB_URI_SINGLE="${MONGODB_URI}/?authMechanism=MONGODB-OIDC&authMechanismProperties=DEVICE_NAME:aws"
 
-echo $MONGODB_URI_SINGLE
-
-export MONGODB_URI="$MONGODB_URI_SINGLE"
 export OIDC_TOKEN_DIR=${OIDC_TOKEN_DIR}
 
-npm run check:oidc
+export MONGODB_URI=${MONGODB_URI:-"mongodb://localhost"}
+
+if [ "$PROVIDER_NAME" = "aws" ]; then
+  export MONGODB_URI_SINGLE="${MONGODB_URI}/?authMechanism=MONGODB-OIDC"
+  export MONGODB_URI_MULTIPLE="${MONGODB_URI}:27018/?authMechanism=MONGODB-OIDC&directConnection=true"
+
+  if [ -z "${OIDC_TOKEN_DIR}" ]; then
+    echo "Must specify OIDC_TOKEN_DIR"
+    exit 1
+  fi
+  npm run check:oidc
+elif [ "$PROVIDER_NAME" = "azure" ]; then
+  if [ -z "${AZUREOIDC_CLIENTID}" ]; then
+    echo "Must specify an AZUREOIDC_CLIENTID"
+    exit 1
+  fi
+  MONGODB_URI="${MONGODB_URI}/?authMechanism=MONGODB-OIDC"
+  MONGODB_URI="${MONGODB_URI}&authMechanismProperties=PROVIDER_NAME:azure"
+  export MONGODB_URI="${MONGODB_URI},TOKEN_AUDIENCE:api%3A%2F%2F${AZUREOIDC_CLIENTID}"
+  npm run check:oidc-azure
```

Comment:
Azure needs to run its prose tests in a separate environment so they are split out.

Reply:
With the GCP/Azure KMS, the tests live in the integration folder but only run on the specialized hosts/environment due to beforeEach skipping. It seems like we're approaching the specialized environment stuff in different ways, I don't feel strongly but should we do away with the `manual/` dir (for this work) in favor of encoding things into mocha?